Tests/StoreKitUnitTests/PriceFormattingRuleSetIntegrationTests.swift

```diff
@@ -0,0 +1,308 @@
+
+        expect(receivedProducts).notTo(beNil())
+        let unwrappedFirstProduct = try XCTUnwrap(receivedProducts?.first)
+        expect(unwrappedFirstProduct.currencyCode) == "RON"
+
+        let unwrappedPriceFormatter = try XCTUnwrap(unwrappedFirstProduct.priceFormatter)
+        XCTAssert(type(of: unwrappedPriceFormatter) == CurrencySymbolOverridingPriceFormatter.self)
+        expect(unwrappedFirstProduct.priceFormatter?.string(from: NSNumber(value: 0))) == "0,00 lei"
+        expect(unwrappedFirstProduct.priceFormatter?.string(from: NSNumber(value: 1))) == "1,00 leu"
+        expect(unwrappedFirstProduct.priceFormatter?.string(from: NSNumber(value: 2))) == "2,00 lei"
+        expect(unwrappedFirstProduct.priceFormatter?.string(from: NSNumber(value: 0.25))) == "0,25 lei"
+    }
+
+    private func testPriceFormatterWithoutRuleSetUsesDefaultFormatter(storeKitVersion: StoreKitVersion) async throws {
+        testSession.locale = Locale(identifier: "ro_RO")
+
+        try await changeStorefront("ROU")
+        let manager = try createSut(storeKitVersion: storeKitVersion, storefront: MockStorefront(countryCode: "ROU"))
+
+        // Create offerings response without priceFormattingRuleSets
+        let offeringsResponse = OfferingsResponse(
+            currentOfferingId: "default",
+            offerings: [.init(
+                identifier: "default",
+                description: "Default offering",
+                packages: [
+                    .init(
+                        identifier: "monthly",
+                        platformProductIdentifier: "com.revenuecat.monthly_4.99.1_week_intro",
+                        webCheckoutUrl: nil
+                    )
+                ],
+                webCheckoutUrl: nil
+            )],
+            placements: nil,
+            targeting: nil,
+            uiConfig: nil,
+            config: .init(priceFormattingRuleSets: [:])
+        )
+
+        mockOfferingsAPI.stubbedGetOfferingsCompletionResult = .success(
+            .init(response: offeringsResponse, httpResponseOriginalSource: .mainServer)
+        )
+
+        // when
+        let result = waitUntilValue { completed in
+            self.offeringsManager.offerings(appUserID: "") {
+                completed($0)
+            }
+        }
+
+        mockDeviceCache.stubbedOfferings = result?.value
+
+        let identifier = "com.revenuecat.monthly_4.99.1_week_intro"
+        var receivedProducts: Set<StoreProduct>?
+
+        receivedProducts = try await manager.products(withIdentifiers: Set([identifier]))
+
+        expect(receivedProducts).notTo(beNil())
+        let unwrappedFirstProduct = try XCTUnwrap(receivedProducts?.first)
+        expect(unwrappedFirstProduct.currencyCode) == "RON"
+
+        let unwrappedPriceFormatter = try XCTUnwrap(unwrappedFirstProduct.priceFormatter)
+        // Should be a regular NumberFormatter
+        XCTAssert(type(of: unwrappedPriceFormatter) == NumberFormatter.self)
```

Comment:
I don't know if check this is really needed or it is going too deep into the specific implementation 🤔 
Wouldn't it be enough just to check the formatted price?